mkosi/tree.py

```diff
@@ -45,6 +48,21 @@ def cp_version() -> GenericVersion:
     return GenericVersion(run(["cp", "--version"], stdout=subprocess.PIPE).stdout.strip().splitlines()[0].split()[3])
 
 
+@contextlib.contextmanager
+def preserve_target_directories_stat(src: Path, dst: Path) -> Iterator[None]:
+    dirs = list(p for d in src.glob("**/") if (dst / (p := d.relative_to(src))).exists())
```

Comment:
Having the `list` call here makes this harder to read due to to the abundance of parens, let's make this a native list comprehension and visually a bit more distinct
```suggestion
    dirs = [p for d in src.glob("**/") if (dst / (p := d.relative_to(src))).exists()]
```